src/lib.rs

```diff
@@ -2498,15 +2498,15 @@ impl<'a> PushFnValueSerializer<'a> {
     /// This consumes `self` to prevent serializing one value multiple times
     pub fn emit<'b, S: 'b + Value>(mut self, s: S) -> Result {
         self.done = true;
-        s.serialize(self.record, self.key, self.serializer)
+        s.serialize(self.record, self.key.clone(), self.serializer)
```

Comment:
Do we have to have `.clone`s here and elsewhere? Can't we pass these keys by value and consume? If the caller has a need to to keep them, they could `.clone` themselves once at the highest layer.

Reply:
I think I need to a proper destruction, moving each field to a local, then passing locals. The borrow checker doesn't like each field being passed by value.